### PR TITLE
Metering: use reconciling framework, use seedsGetter more consistently

### PR DESCRIPTION
**Pull request by xrstf**
**What this PR does / why we need it**:
This does some quality of life changes to the new metering code.

* no more hardcoded "kubermatic" namespace
* use reconciling to manage secret
* use seedsGetter in more places

/kind cleanup

**Does this PR introduce a user-facing change? Then add your Release Note here**:
```release-note
NONE
```

### CODE REVIEW

File: pkg/ee/metering/reconcile.go [hunk 1]
```diff
@@ -238,8 +237,8 @@ func undeploy(ctx context.Context, client ctrlruntimeclient.Client, namespace st
 }
 
 // cleanupLegacyMeteringResources removes all active parts of the legacy metering installation.
-func cleanupLegacyMeteringResources(ctx context.Context, client ctrlruntimeclient.Client) error {
-	key := types.NamespacedName{Namespace: resources.KubermaticNamespace, Name: meteringToolName}
+func cleanupLegacyMeteringResources(ctx context.Context, client ctrlruntimeclient.Client, namespace string) error {
```
Comment (mate4st) [inline pkg/ee/metering/reconcile.go:245]: We need to keep the hardcoded namespace here, as the legacy components were always deployed to kubermatic namespace. 

Ref: https://github.com/kubermatic/kubermatic/blob/release/v2.20/pkg/ee/metering/reconcile.go#L53

**Review by mate4st — CHANGES_REQUESTED** [review]

File: pkg/ee/metering/reconcile.go (hunk 1 above)
Comment (xrstf) [reply pkg/ee/metering/reconcile.go:245]: Good catch!

**Review by ahmedwaleedmalik — APPROVED** [review]
/approve ⚡ 
/hold so that @WeirdMachine can take another look as well

**Review by mate4st — APPROVED** [review]

**Pull request merged**